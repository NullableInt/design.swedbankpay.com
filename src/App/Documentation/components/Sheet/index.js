import React, { Component } from "react";

import { ComponentPreview, DocToc, Icon, Property } from "#";
import { sheet } from "$/px-script/main";

const Example = () => (
    <>
        <h2 id="example">Example</h2>
        <p>Example sheet...</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="sheet" id="demo-sheet">
                <section>{"\n"}
                    <a href="#" className="sheet-close">{"\n\t\t\t"}
                        <Icon icon="close" />{"\n\t\t"}
                    </a>
                    <h2>Sheet component</h2>
                    <p>This is the sheet component. It can contain anything you want.</p>
                    <p>It can also contain lots of useless text, just like the text in this paragraph. Or it can contain useful info and cheats. For example if you have a math quiz coming up;</p>
                    <ul className="list">
                        <li><code>3 + 8 = 11</code></li>
                        <li><code>36 - 22 = 14</code></li>
                        <li><code>256 + 113 = 369</code></li>
                        <li><code>356 - 1 = 355</code></li>
                    </ul>
                    <hr/>
                    <form onSubmit={e => e.preventDefault()}>
                        <div className="form-group">
                            <label htmlFor="full-name">Full name</label>
                            <div className="input-group">
                                <input type="text" className="form-control" id="full-name" placeholder="Enter your full name" />
                            </div>
                            <div className="help-block">Your full name</div>
                        </div>
                        <div className="form-group">
                            <label htmlFor="email-address">Email</label>
                            <div className="input-group">
                                <input type="text" className="form-control" id="email-address" placeholder="Enter your email" />
                            </div>
                            <div className="help-block">Email means electronic mail, not express mail</div>
                        </div>
                        <div className="form-group">
                            <div className="checkbox">
                                <input type="checkbox" id="subscribe-checkbox" />
                                <label htmlFor="subscribe-checkbox">Sign up to our math newsletter</label>
                            </div>
                        </div>
                        <button className="btn btn-primary" type="submit" onClick={e => e.preventDefault()}>Submit</button>
                    </form>
                </section>
            </div>{"\n"}
            <button className="btn btn-primary" type="button" data-sheet-open="demo-sheet">Open sheet</button>
        </ComponentPreview>
    </>
);

const UsageWithJavascript = () => (
    <>
        <h2 id="usage-with-javascript">Usage with javascript</h2>
        <p>To programmatically open a sheet, use <Property value={"px.sheet.open(\"SHEET_ID\")"} />.</p>
        <p>To programmatically close a sheet, use <Property value={"px.sheet.close(\"SHEET_ID\")"} />.</p>
    </>
);

const SheetText = () => (
    <div className="col-lg-10 doc-body">
        <p className="lead">Sheet...</p>
        <Example />
        <UsageWithJavascript />
    </div>
);

class Sheet extends Component {
    componentDidMount () {
        sheet.init();
    }

    render () {
        return (
            <div className="doc-container">
                <div className="row">
                    <SheetText />
                    <DocToc component={SheetText} />
                </div>
            </div>
        );
    }
}

export default Sheet;

/* For testing */
export { Example, UsageWithJavascript, SheetText };
