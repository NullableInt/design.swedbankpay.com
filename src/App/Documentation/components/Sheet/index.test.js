import React from "react";
import { shallow } from "enzyme";

import Sheet, { Example, UsageWithJavascript, SheetText } from "./index";

describe("Documentation: Sheet", () => {
    it("is defined", () => {
        expect(Sheet).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Sheet />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Example", () => {
        it("is defined", () => {
            expect(Example).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Example />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("UsageWithJavascript", () => {
        it("is defined", () => {
            expect(UsageWithJavascript).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<UsageWithJavascript />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SheetText", () => {
        it("is defined", () => {
            expect(SheetText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SheetText />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
