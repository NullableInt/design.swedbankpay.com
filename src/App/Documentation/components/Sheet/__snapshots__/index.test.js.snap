// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: Sheet Example onClick prevents default 1`] = `
<Fragment>
  <h2
    id="example"
  >
    Example
  </h2>
  <p>
    Example sheet...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="sheet"
      id="demo-sheet"
    >
      <section>
        

        <a
          className="sheet-close"
          href="#"
        >
          
			
          <Icon
            icon="close"
          />
          
		
        </a>
        <h2>
          Sheet component
        </h2>
        <p>
          This is the sheet component. It can contain anything you want.
        </p>
        <p>
          It can also contain lots of useless text, just like the text in this paragraph. Or it can contain useful info and cheats. For example if you have a math quiz coming up;
        </p>
        <ul
          className="list"
        >
          <li>
            <code>
              3 + 8 = 11
            </code>
          </li>
          <li>
            <code>
              36 - 22 = 14
            </code>
          </li>
          <li>
            <code>
              256 + 113 = 369
            </code>
          </li>
          <li>
            <code>
              356 - 1 = 355
            </code>
          </li>
        </ul>
        <hr />
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="full-name"
            >
              Full name
            </label>
            <div
              className="input-group"
            >
              <input
                className="form-control"
                id="full-name"
                placeholder="Enter your full name"
                type="text"
              />
            </div>
            <div
              className="help-block"
            >
              Your full name
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="email-address"
            >
              Email
            </label>
            <div
              className="input-group"
            >
              <input
                className="form-control"
                id="email-address"
                placeholder="Enter your email"
                type="text"
              />
            </div>
            <div
              className="help-block"
            >
              Email means electronic mail, not express mail
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="checkbox"
            >
              <input
                id="subscribe-checkbox"
                type="checkbox"
              />
              <label
                htmlFor="subscribe-checkbox"
              >
                Sign up to our math newsletter
              </label>
            </div>
          </div>
          <button
            className="btn btn-primary"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </form>
      </section>
    </div>
    

    <button
      className="btn btn-primary"
      data-sheet-open="demo-sheet"
      type="button"
    >
      Open sheet
    </button>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Sheet Example onSubmit prevents default 1`] = `
<Fragment>
  <h2
    id="example"
  >
    Example
  </h2>
  <p>
    Example sheet...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="sheet"
      id="demo-sheet"
    >
      <section>
        

        <a
          className="sheet-close"
          href="#"
        >
          
			
          <Icon
            icon="close"
          />
          
		
        </a>
        <h2>
          Sheet component
        </h2>
        <p>
          This is the sheet component. It can contain anything you want.
        </p>
        <p>
          It can also contain lots of useless text, just like the text in this paragraph. Or it can contain useful info and cheats. For example if you have a math quiz coming up;
        </p>
        <ul
          className="list"
        >
          <li>
            <code>
              3 + 8 = 11
            </code>
          </li>
          <li>
            <code>
              36 - 22 = 14
            </code>
          </li>
          <li>
            <code>
              256 + 113 = 369
            </code>
          </li>
          <li>
            <code>
              356 - 1 = 355
            </code>
          </li>
        </ul>
        <hr />
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="full-name"
            >
              Full name
            </label>
            <div
              className="input-group"
            >
              <input
                className="form-control"
                id="full-name"
                placeholder="Enter your full name"
                type="text"
              />
            </div>
            <div
              className="help-block"
            >
              Your full name
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="email-address"
            >
              Email
            </label>
            <div
              className="input-group"
            >
              <input
                className="form-control"
                id="email-address"
                placeholder="Enter your email"
                type="text"
              />
            </div>
            <div
              className="help-block"
            >
              Email means electronic mail, not express mail
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="checkbox"
            >
              <input
                id="subscribe-checkbox"
                type="checkbox"
              />
              <label
                htmlFor="subscribe-checkbox"
              >
                Sign up to our math newsletter
              </label>
            </div>
          </div>
          <button
            className="btn btn-primary"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </form>
      </section>
    </div>
    

    <button
      className="btn btn-primary"
      data-sheet-open="demo-sheet"
      type="button"
    >
      Open sheet
    </button>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Sheet Example renders 1`] = `
<Fragment>
  <h2
    id="example"
  >
    Example
  </h2>
  <p>
    Example sheet...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="sheet"
      id="demo-sheet"
    >
      <section>
        

        <a
          className="sheet-close"
          href="#"
        >
          
			
          <Icon
            icon="close"
          />
          
		
        </a>
        <h2>
          Sheet component
        </h2>
        <p>
          This is the sheet component. It can contain anything you want.
        </p>
        <p>
          It can also contain lots of useless text, just like the text in this paragraph. Or it can contain useful info and cheats. For example if you have a math quiz coming up;
        </p>
        <ul
          className="list"
        >
          <li>
            <code>
              3 + 8 = 11
            </code>
          </li>
          <li>
            <code>
              36 - 22 = 14
            </code>
          </li>
          <li>
            <code>
              256 + 113 = 369
            </code>
          </li>
          <li>
            <code>
              356 - 1 = 355
            </code>
          </li>
        </ul>
        <hr />
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="full-name"
            >
              Full name
            </label>
            <div
              className="input-group"
            >
              <input
                className="form-control"
                id="full-name"
                placeholder="Enter your full name"
                type="text"
              />
            </div>
            <div
              className="help-block"
            >
              Your full name
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="email-address"
            >
              Email
            </label>
            <div
              className="input-group"
            >
              <input
                className="form-control"
                id="email-address"
                placeholder="Enter your email"
                type="text"
              />
            </div>
            <div
              className="help-block"
            >
              Email means electronic mail, not express mail
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="checkbox"
            >
              <input
                id="subscribe-checkbox"
                type="checkbox"
              />
              <label
                htmlFor="subscribe-checkbox"
              >
                Sign up to our math newsletter
              </label>
            </div>
          </div>
          <button
            className="btn btn-primary"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </form>
      </section>
    </div>
    

    <button
      className="btn btn-primary"
      data-sheet-open="demo-sheet"
      type="button"
    >
      Open sheet
    </button>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Sheet UsageWithJavascript renders 1`] = `
<Fragment>
  <h2
    id="usage-with-javascript"
  >
    Usage with javascript
  </h2>
  <p>
    To programmatically open a sheet, use 
    <Property
      value="px.sheet.open(\\"SHEET_ID\\")"
    />
    .
  </p>
  <p>
    To programmatically close a sheet, use 
    <Property
      value="px.sheet.close(\\"SHEET_ID\\")"
    />
    .
  </p>
</Fragment>
`;

exports[`Documentation: Sheet renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Sheet...
  </p>
  <Example />
  <UsageWithJavascript />
</DocContainer>
`;
